src/app/components/chat/client.ts

```diff
@@ -157,18 +182,29 @@ function reset(chat: ChatClient) {
 }
 
 async function connect(chat: ChatClient) {
+	const chatId = ChatClient.nextId;
+	chat.id = chatId;
+	ChatClient.nextId = (ChatClient.nextId + 1) % 100;
```

Comment:
May I ask why?

Reply:
tracking the chat id as it is reinstantiated.
if it changes, thats how i super hackily abort the rest of the connection flow and the pollRequests in the now outdated instance.